packages/ERTP/src/safeMath.js

```diff
@@ -1,13 +1,13 @@
 import { Nat } from '@endo/nat';
 
-/** @typedef {(x: number | bigint, y: number | bigint) => NatValue} NatOp */
+/** @typedef {(x: number | bigint, y: number | bigint) => bigint} NatOp */
```

Comment:
Does typescript have a type meaning bigint >= 0n? Or, can such a type be expressed in typescript?

Reply:
Can't be expressed in TypeScript. Very few languages can express such Refinement Types. Per GPT 4o,
> Languages or tools that support refinement types include:
>	•	Liquid Haskell
>	•	F (F star)*
>	•	Idris (which blends dependent and refinement typing)
>	•	Dafny
>	•	Why3

Reply:
Pascal had subrange types ;)

I think Spark ADA had fancy subrange types which AFAIK might be refinement types.